File: Problems/Leetcode/contest/BiWeekly/2.cpp
#include <bits/stdc++.h>

using namespace std;

static constexpr int MOD = 1e9 + 7;

class Solution {
public:
    int minFlips(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        int rc = 0, cc = 0;

        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n / 2; ++j) {
                if (grid[i][j] != grid[i][n - 1 - j]) {
                    rc++;
                }
            }
        }

        for (int j = 0; j < n; ++j) {
            for (int i = 0; i < m / 2; ++i) {
                if (grid[i][j] != grid[m - 1 - i][j]) {
                    cc++;
                }
            }
        }

        return min(rc, cc);
    }
};

int main() {
    Solution sol;

    return 0;
}

File: Problems/Leetcode/contest/BiWeekly/2.py
from typing import List

fmax = lambda x, y: x if x > y else y
fmin = lambda x, y: x if x < y else y
MOD = 1_000_000_007


class Solution:
    def minFlips(self, grid: List[List[int]]) -> int:
        n, m = len(grid), len(grid[0])
        return fmin(
            sum(
                1
                for i in range(n)
                for j in range(m // 2)
                if grid[i][j] != grid[i][m - 1 - j]
            ),
            sum(
                1
                for j in range(m)
                for i in range(n // 2)
                if grid[i][j] != grid[n - 1 - i][j]
            ),
        )
